sources/packages/api/src/services/education-program-offering/education-program-offering.service.ts

```diff
@@ -673,17 +707,208 @@ export class EducationProgramOfferingService extends RecordDataModelService<Educ
       .where("application.applicationStatus != :cancelledStatus", {
         cancelledStatus: ApplicationStatus.cancelled,
       })
-      .andWhere("offering.id = :offeringId")
-      .setParameters({
-        cancelledStatus: ApplicationStatus.cancelled,
-        offeringId,
-      })
-      .groupBy("offering.id");
+      .andWhere("offering.id = :offeringId", { offeringId });
 
     const precedingOffering = await query.getRawOne();
 
     return {
       applicationsCount: +precedingOffering?.applicationsCount,
     };
   }
+
+  /**
+   * Get applications that are impacted by an offering change
+   * to submit reassessment.
+   * @param offeringId offering requested for change.
+   * @returns applications.
+   */
+  async getApplicationsToSubmitReassessment(
+    offeringId: number,
+  ): Promise<ApplicationAssessmentSummary[]> {
+    const applications = await this.dataSource
+      .getRepository(Application)
+      .createQueryBuilder("application")
+      .select("application.id")
+      .addSelect("assessment.id")
+      .addSelect("assessment.assessmentData IS NULL", "hasAssessmentData")
+      .addSelect("application.data->>'workflowName'", "workflowName")
+      .addSelect("assessment.assessmentWorkflowId", "assessmentWorkflowId")
+      .innerJoin("application.currentAssessment", "assessment")
+      .innerJoin(
+        EducationProgramOffering,
+        "offering",
+        "offering.precedingOffering.id = assessment.offering.id",
+      )
+      .where(
+        "application.applicationStatus NOT IN (:...invalidOfferingChangeStatus)",
+        {
+          invalidOfferingChangeStatus: [
+            ApplicationStatus.cancelled,
+            ApplicationStatus.overwritten,
```

Comment:
Was this condition also added to the method doing the count of the impacted ones?

Reply:
No. I missed it. Adding. 👍